docs/role-based-access-control.rst

```diff
@@ -0,0 +1,188 @@
+Role Based Access Control
+=========================
+
+Role based access control in Pulp Container is configured using Access Policies for the following
+``viewset_names``:
+
+* ``pulp_container/namespaces``
+* ``distributions/container/container``
+* ``repositories/container/container-push``
+* ``remotes/container/container``
+* ``repositories/container/container``
+* ``repositories/container/container-push/versions``
+* ``repositories/container/container/versions``
+
+This document describes the default access policies shipped with Pulp Container. Each of the above
+policies can be modified to achieve a different RBAC behavior.
+
+Repositories that are created using ``podman push`` or ``docker push`` are considered public and anyone
+can ``pull`` the images in them. See below about creating private repositories.
+
+Namespaces
+----------
+
+Pulp Container namespaces allow users to reuse repository names under different context. The
+namespace can represent an organization, a team, or any other kind of logical grouping of container
+repositories. Namespaces provide a naming convention for container repositories. Repositories in
+the ``foo`` namespace are named ``foo/something`` and ``foo/something-else``.
+
+The default access policy for ``pulp_container/namespaces`` requires a user to have the
+``container.add_containernamespace`` permission to create a new namespace. The new namespace can be
+created by pushing an image using ``podman`` or ``docker`` client. This same permission allows the user
+of Pulp's API to create a new namespace.
+
+The creation of a new namespace creates three user groups that can access the namespace:
+Owners, Collaborators, and Consumers. The user that creates the namespace is automatically added to
+the Owners group.
+
+Namespace Owners
+~~~~~~~~~~~~~~~~
+
+The group name is ``container.namespace.owners.<namespace name>``. This group has the following
+object permissions for the namespace::
+
+     "container.view_containernamespace"
+     "container.delete_containernamespace"
+     "container.namespace_add_containerdistribution",
+     "container.namespace_delete_containerdistribution
+     "container.namespace_view_containerdistribution"
+     "container.namespace_pull_containerdistribution"
+     "container.namespace_push_containerdistribution"
+     "container.namespace_change_containerdistribution"
+     "container.namespace_view_containerpushrepository"
+     "container.namespace_modify_content_containerpushrepository"
+
+The users in the owners group have the permissions to add/remove users from all three groups
+associated with the namespace. They also have the ability to create, update, and delete
+repositories in the namespace.
```

Comment:
should we expand here and clarify that in order to delete 'a registry repo' a container_distribution should be removed via pulp api?